feature-assets/src/main/java/io/novafoundation/nova/feature_assets/presentation/novacard/topup/TopUpCardViewModel.kt

```diff
@@ -0,0 +1,184 @@
+import io.novafoundation.nova.feature_wallet_api.presentation.mixin.amountChooser.setAmount
+import io.novafoundation.nova.feature_wallet_api.presentation.mixin.fee.FeeLoaderMixin
+import io.novafoundation.nova.feature_wallet_api.presentation.mixin.fee.SimpleGenericFee
+import io.novafoundation.nova.feature_wallet_api.presentation.mixin.fee.awaitDecimalFee
+import io.novafoundation.nova.feature_wallet_api.presentation.mixin.fee.commissionAsset
+import io.novafoundation.nova.feature_wallet_api.presentation.mixin.fee.connectGenericWith
+import io.novafoundation.nova.feature_wallet_api.presentation.mixin.fee.createGenericChangeableFee
+import io.novafoundation.nova.runtime.multiNetwork.ChainRegistry
+import io.novafoundation.nova.runtime.multiNetwork.chainWithAsset
+import kotlinx.coroutines.flow.MutableStateFlow
+import kotlinx.coroutines.flow.first
+import kotlinx.coroutines.flow.flatMapLatest
+import kotlinx.coroutines.flow.map
+import kotlinx.coroutines.launch
+
+class TopUpCardViewModel(
+    private val chainRegistry: ChainRegistry,
+    private val interactor: WalletInteractor,
+    private val sendInteractor: SendInteractor,
+    private val router: AssetsRouter,
+    private val payload: TopUpCardPayload,
+    private val validationExecutor: ValidationExecutor,
+    private val resourceManager: ResourceManager,
+    feeLoaderMixinFactory: FeeLoaderMixin.Factory,
+    selectedAccountUseCase: SelectedAccountUseCase,
+    amountChooserMixinFactory: AmountChooserMixin.Factory,
+    addressInputMixinFactory: AddressInputMixinFactory,
+) : BaseViewModel(), Validatable by validationExecutor {
+
+    private val chainWithAssetFlow = flowOf { chainRegistry.chainWithAsset(payload.asset.chainId, payload.asset.chainAssetId) }
+
+    private val chainFlow = chainWithAssetFlow.map { it.chain }
+    private val chainAssetFlow = chainWithAssetFlow.map { it.asset }
+
+    private val selectedAccount = selectedAccountUseCase.selectedMetaAccountFlow()
+        .inBackground()
+        .share()
+
+    private val sendInProgressFlow = MutableStateFlow(false)
+
+    private val assetFlow = chainAssetFlow.flatMapLatest(interactor::assetFlow)
+        .shareInBackground()
+
+    val addressInputMixin = with(addressInputMixinFactory) {
+        create(
+            inputSpecProvider = singleChainInputSpec(chainFlow),
+            errorDisplayer = this@TopUpCardViewModel::showError,
+            showAccountEvent = null,
+            coroutineScope = this@TopUpCardViewModel,
+        )
+    }
+
+    val amountChooserMixin: AmountChooserMixin.Presentation = amountChooserMixinFactory.create(
+        scope = this,
+        assetFlow = assetFlow,
+        balanceLabel = R.string.wallet_balance_transferable,
+        balanceField = Asset::transferable,
+    )
+
+    val feeMixin = feeLoaderMixinFactory.createGenericChangeableFee<OriginGenericFee>(assetFlow, coroutineScope)
```

Comment:
I use OriginFee here to avoid refactoring in send flow since OriginFee is required in validation